### ISSUE TITLE
[Bug]: 30.0.4+ The "Update All" button to update all apps does not work properly if I am already logged in.

### ISSUE THREAD

**Issue by tigernero79** [nextcloud/server#49866]
### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Bug Description
On NC 30.0.4

When updates are available for several apps, an "Update All" button appears on the Apps > Updates page. However, clicking this button works if you are just logged in, but if you open your nextcloud that is already logged in it does not work, you have to click update all for each application that needs to be updated. Absurd behavior. In practice, if you are already logged in and click the update all button and 4 updates are available you have to click the update all button 4 times. If you log out and log back in, everything works, you just have to click update all once.

### Steps to reproduce

1. go to the updates section while already logged in (you are not logged in)
2. Click on the update all item
3. you will notice that you have to click the update all button as many times as there are applications to update
4. this does not happen if you log in and then go to update, in that case everything works, just click update all once

### Expected behavior

I would expect that clicking the "update all" button only once would update all applications even if I'm already logged in

### Nextcloud Server version

30

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```json

```

### List of activated Apps

```shell

```

### Nextcloud Signing status

```shell

```

### Nextcloud Logs

```json

```

**Comment by tigernero79** [comment]
a similar issue but that has been closed is here:

https://github.com/nextcloud/server/issues/49144

only that it brings a new bug in nextcloud 30.0.4

**Comment by bpmartin20** [comment]
Still a problem on 31.0.6

**Comment by szaimen** [comment]
Hi, I think this was recently fixed. Can you please try again with one of the latest releases and report back if this is fixed? Thanks in advance!

**Comment by tigernero79** [comment]
the latest version nextcloud 32.0.5 still has this problem if you are already logged in and try to update more than one application, if instead you log in and then try to update the problem is not there

**Comment by szaimen** [comment]
I just checked and it seems we forgot to backport: 
- https://github.com/nextcloud/server/pull/56923
- https://github.com/nextcloud/server/pull/57903

**Comment by tigernero79** [comment]
so we have to wait for next release for fix right?

**Comment by indivisualvj** [comment]
- i have multiple apps that can be updated.
- i am logged in
- click on Update All and then have to validate that i am admin
- only the app at the very bottom get's updated

**Comment by tigernero79** [comment]
yes but if you click update again, you will see that then it will update all the other apps that it hasn't updated

**Comment by bpmartin20** [comment]
True, but it's still a bug. I hope it gets fixed soon.

**Comment by tigernero79** [comment]
sure, I was the one who opened the bug

**Comment by ant0nwax** [comment]
Update all [two apps] happend again here on-premise that after password authentication only one app was updated. If QA are curious about help in testing, please respect that this never worked correctly here since at least a couple of months or a year. If someone has time to explain, how to setup debug log collection and switch debug off before and after touching update all, do so. br.

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by bpmartin20** [comment]
This is still a problem on 32.0.6

**Comment by Lawkss** [comment]
Years later this is still not fixed. Most annoying bug. Always have to hit update all button twice because it only updates 1 app instead of all.

**Comment by bpmartin20** [comment]
As of about 2 weeks ago, this was suddenly fixed for me. I stay on current 
maintenance,  but I don't do anything special ( no early releases, etc). 
I'm using Firefox, though I don't know if that's relevant. Good luck.

**Comment by tigernero79** [comment]
Problem solved for me on 33.0.0 nextcloud

**Comment by Lawkss** [comment]
Worked first time on 32.0.7 will observe.

**Comment by indivisualvj** [comment]
solved for me on 33.0.1

**Comment by szaimen** [comment]
Thanks so I guess we can close this finally

**Comment by undrwater** [comment]
This has showed again on Nextcloud Hub 26 Spring 34.0.1.

Apps | Updates | Update all Apps  --> authentication screen | add password | authentication screen returns asking for password now with red text, but no error about wrong password (it was correct anyway). 

The app updates anyway, but one must refresh or navigate away from the update page, then navigate back.

Nothing in logs.
